### PR TITLE
[tests-only][full-ci] add test to list shares shared to federated user

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds tests where the sharer lists the shares shared by him to a federated user.
``` feature
Scenario: sharer lists the shares shared by him to a federated user
```

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/10073
- Bug Report: https://github.com/owncloud/ocis/issues/9908

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiOcm/share.feature [hunk 1]
```diff
@@ -288,3 +288,139 @@ Feature: an user shares resources usin ScienceMesh application
...
       | Manager          |
+
+  @issue-9908
+  Scenario: sharer lists the shares shared by him to a federated user
+    Given using server "LOCAL"
+    And "Alice" has created the federation share invitation
+    And using server "REMOTE"
+    And "Brian" has accepted invitation
+    And using server "LOCAL"
+    And user "Alice" has uploaded file with content "ocm test" to "/textfile.txt"
```
Comment (anon-pradip) [inline tests/acceptance/features/apiOcm/share.feature:299]: IMHO, placing the step:
` And user "Alice" has uploaded file with content "ocm test" to "/textfile.txt"` just below the `Given step` sounds more better

File: tests/acceptance/features/apiOcm/share.feature [hunk 2]
```diff
@@ -288,3 +288,139 @@ Feature: an user shares resources usin ScienceMesh application
...
+    And user "Alice" has sent the following resource share invitation to federated user:
+      | resource        | textfile.txt                  |
+      | space           | Personal                      |
+      | sharee          | Brian                         |
+      | shareType       | user                          |
+      | permissionsRole | Viewer                        |
+      | federatedServer | @federation-ocis-server:10200 |
+    When using server "LOCAL"
+    And user "Alice" lists the shares shared by him using the Graph API
```
Comment (anon-pradip) [inline tests/acceptance/features/apiOcm/share.feature:308]: We are already in `LOCAL` server so I think  we can skip `When using server "LOCAL"`
```suggestion
    When user "Alice" lists the shares shared by him using the Graph API
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 3]
```diff
@@ -288,3 +288,138 @@ Feature: an user shares resources usin ScienceMesh application
...
+    And "Brian" has accepted invitation
+    And using server "LOCAL"
+    And user "Alice" has sent the following resource share invitation to federated user:
+      | resource        | textfile.txt                  |
+      | space           | Personal                      |
+      | sharee          | Brian                         |
+      | shareType       | user                          |
+      | permissionsRole | Viewer                        |
+      | federatedServer | @federation-ocis-server:10200 |
+    When user "Alice" lists the shares shared by him using the Graph API
```
Comment (anon-pradip) [inline tests/acceptance/features/apiOcm/share.feature:307]:
```suggestion
    When user "Alice" lists the shares shared by her using the Graph API
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 4]
```diff
@@ -288,3 +288,138 @@ Feature: an user shares resources usin ScienceMesh application
...
+              "id"
+            ],
+            "properties": {
+              "driveId": {
+                "type": "string",
+                "pattern": "^%space_id_pattern%$"
+              },
+              "driveType": {
+                "type": "string",
+                "enum": ["personal"]
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiOcm/share.feature:336]:
```suggestion
              "driveType": {
                "const": "personal"
```

Similar in other properties as well.

**Review by PrajwolAmatya — APPROVED** [review]
LGTM

File: tests/acceptance/features/apiOcm/share.feature [hunk 5]
```diff
@@ -288,3 +288,125 @@ Feature: an user shares resources usin ScienceMesh application
       | Space Viewer     |
       | Space Editor     |
       | Manager          |
+
+  @issue-9908
+  Scenario: sharer lists the shares shared by him to a federated user
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:293]:
```suggestion
  Scenario: sharer lists the shares shared to a federated user
```

**Review by saw-jan — COMMENTED** [review]
others look good

**Review by saw-jan — APPROVED** [review]

**Pull request merged**